### ISSUE TITLE
Make CxAnalytix serverless

### ISSUE THREAD

**Issue by istvanfedak** [checkmarx-ts/CxAnalytix#162]
### Describe the problem

> Currently CxAnalytix is not serverless and needs to be run on an instance.

### Proposed solution

> Provide a CxAnalytix public lambda layer that would grant an AWS Lambda access to run CxAnalytix. See [Klayers](https://github.com/keithrozario/Klayers) as an example of public layers that can be used by Lambda.

### Additional details

> This would allow extract transform and load jobs to run serverless and would remove the overhead of maintaining an instance.

Ideally the reporting generated by `CxAnalytix` should be available as endpoints of the Checkmarx API

**Comment by nleach999** [comment]
Hi @istvanfedak ,

Can you please expand on your use-case a bit more?  CxAnalytix crawls typically take more time than is feasible to put in a Lambda.  There is also no external input; it chats with the SAST API, downloads scans in scope, outputs records.

There is a [Docker container](https://github.com/checkmarx-ts/CxAnalytix/pkgs/container/cxanalytix%2Fcxanalytix) available that can run without a full machine instance.  

If you're looking for something that invokes Lambda functions with the transformed data messages, the [AMQP](https://github.com/checkmarx-ts/CxAnalytix/wiki/AMQP) output mechanism can [invoke Lambda functions](https://aws.amazon.com/about-aws/whats-new/2021/07/aws-lambda-now-supports-amazon-mq-for-rabbitmq-as-an-event-source/) when events are received at an AMQP endpoint.  (This might need a bit of work to send boundary marker messages so async workflows can orchestrate events properly, but that might depend on your use-case.)

**Comment by istvanfedak** [comment]
Hi @nleach999,

I hope all is well. I'll look into the Docker container.

I was wondering if it was feasible to have CxAnalytix available as a AWS Lambda layer and be able to provide all the scan information for a single given scan (scanId). This would provide the AWS Lambda with a pre-packaged CxAnalytix executable that can be invoked from the Lambda.

From there you could split the workload between multiple lambdas using a fan out approach. For example, one lambda would  get a list of all the projects in Checkmarx (this can be done using the SAST api as well) and it would invoke a Lambda project handler per project. From there the project handler Lambda would invoke CxAnalytix to obtain all the latest scan information for that specific project and save it in either a database or an S3 (the customer would decide what to do with the data). We need to pull the historical label data for a scan issue and we were hoping to leverage CxAnalytix.

We built a similar workflow using the Checkmarx API. We have a Lambda that gets a list of projects and then  invokes a project handler lambda per project. The project handler lambda gets all the latest scan information, generates a CSV scan report and parses the data (not to mention the back end batch job to delete the reports on the Checkmarx server). The only thing we can't obtain is the historical label data. There is an endpoint in the Checkmarx API that allows us to pull the label for a scan issue (GET /sast/scans/{scanId}/results/{pathId}/labels) but it doesn't provide the historical data. 

The scan report issue label historical data is all compressed together and it's quite hard to parse out. This would be a sample of the scan issue historical data we're trying to obtain:

```json
[
  {
    "state": 0,
    "severity": 0,
    "userAssignment": "admin",
    "comment": "string",
    "datetime": "string"
  },
  {
    "state": 0,
    "severity": 0,
    "userAssignment": "admin",
    "comment": "string",
    "datetime": "string"
  }
]
```

Thanks for the help!

**Comment by istvanfedak** [comment]
@nleach999 yes we ran into the same scaling issues and we limited the number of concurrent lambda executions down to 5 lambdas running concurrently. AWS Lambda has a built in event queue so the Lambda that got the projects would add events to the project handler lambda queue. 

We're not looking to pull the data live and we want to use it more for reporting or analytics. This ETL job runs once per day.

Since `comments come packed in a single field that you'd have to parse even if CxAnalytix were extracting it`, we won't be able to leverage `CxAnalytix`.

Thanks you for the explanation it was very helpful!
